protocol/messenger_handler.go

```diff
@@ -1379,7 +1379,16 @@ func (m *Messenger) matchChatEntity(chatEntity common.ChatEntity) (*Chat, error)
 		if err != nil {
 			return nil, err
 		}
-		if !emojiReaction && !canPost {
+
+		community, err := m.communitiesManager.GetByIDString(chat.CommunityID)
+		if err != nil {
+			return nil, err
+		}
+
+		isMemberAdmin := community.IsMemberAdmin(chatEntity.GetSigPubKey())
+		pinMessageAllowed := community.AllowsAllMembersToPinMessage()
+
+		if (!isMemberAdmin && !pinMessageAllowed) || (!emojiReaction && !canPost) {
```

Comment:
Should there be an additionnal check on pin? It seems like this might block people from posting normal messages as well.
Something like 
```go
(isPinMessage && !isMemberAdmin && !pinMessageAllowed)
```

Where `isPinMessage` comes from a new `case` in the `switch` above.

Reply:
Ah, forgot to add that. In my head I was still inside `HandlePinMessage`...

Thanks man, will update